client/packages/common/src/ui/components/inputs/TextInput/NumbericTextInput.test.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { NumericTextInput, NumericTextInputProps } from './NumericTextInput';
+import { TestingProvider } from '@common/utils';
+
+describe('Test NumericTextInput component', () => {
+  const TestNumericTextInput = (props: NumericTextInputProps) => {
+    const [value, setValue] = useState<number | undefined>();
+    return (
+      <TestingProvider>
+        <NumericTextInput
+          value={value}
+          onChange={value => setValue(value)}
+          {...props}
+        />
+      </TestingProvider>
+    );
+  };
+
+  it('should handle decimal number input', async () => {
+    const { getByRole } = render(<TestNumericTextInput decimalLimit={10} />);
+    const input = getByRole('textbox');
+
+    fireEvent.change(input, { target: { value: '1.' } });
+    expect(input).toHaveValue('1.');
+    fireEvent.blur(input);
+    expect(input).toHaveValue('1');
```

Comment:
This fixes it for me. The value is not propagated in onBlur...

```
onBlur={() => {
          const v = Number(parse(textValue)) || undefined;
          onChange(v);
          setTextValue(format(v, { minimumFractionDigits: decimalMin }));
        }}
```

Note, not sure if I like this NaN check `|| undefined` since you have to know what it is doing...

Reply:
That's odd -- the reason I had no `onChange` call as part of the `onBlur` is that it shouldn't be needed. The only difference between the text value when it loses focus and the last `onChange` update is characters that wouldn't affect the number value (i.e. trailing `0`s or `.`). 

(Using that same reasoning, you don't actually need the `||undefined` check, since the text value at this stage should never not parse to a valid number)

Do you know why your test isn't getting the correct value? It should be displaying the `textValue` right? I'm slightly reluctant to add something that causes an extra render just to make a test pass.

Reply:
Its actually a bug. You can reproduce it in storybook:

1) put in a 20
2) copy and paste the "1." in (don't type it in).
3) blur
4) it will then revert to the previous value 20 because onChange never got called...

If you type it in the onChange(1) will be called before you type the "."  and then it reverts to 1. and you don't notice the bug.

You need the `|| undefined` if you have an empty string in the text field while doing the blur...

While playing around I found another bug where you can't delete a single ".". I added another test case for it.

Reply:
Okay, thanks, will fix this morn

Reply:
Okay @clemens-msupply, have added that and the fix for the single decimal. d528a85c7

I tried to add another test for the single decimal, but I couldn't get it to work as I needed to simulate the "Backspace" key to actually test it properly. However, the `fireEvent.keyDown` or `.keyPress` methods didn't seem to work. Probably don't really need it, but if you have any idea how to test it, let me know?

Reply:
No sorry haven't done this yet, fireEvent.keyDown sounds right to me but might be wrong...